packages/core/src/operations.ts

```diff
@@ -229,9 +230,18 @@ export class TxOperations implements Omit<Client, 'notify'> {
     const hierarchy = this.client.getHierarchy()
     const mixClass = Hierarchy.mixinOrClass(doc)
     if (hierarchy.isMixin(mixClass)) {
-      // TODO: Rework it is wrong, we need to split values to mixin update and original document update if mixed.
+      const { docUpdate, mixinUpdate } = this.splitMixinUpdate(update, mixClass)
```

Comment:
Let's use apply to pass them all to server and execute instead of multiple requests.

Reply:
Fixed